File: solutions/go/isogram/1/isogram.go
package isogram

import (
	"strings"
	"unicode"
)

func IsIsogram(word string) bool {
	word = strings.ToLower(word)
	seen := map[rune]int{}

	for _, char := range word {
		if !unicode.IsLetter(char) {
			continue
		}
		if _, ok := seen[char]; ok {
			return false
		}
		seen[char]++
	}

	return true
}

File: solutions/javascript/isogram/1/isogram.js
//
// This is only a SKELETON file for the 'Isogram' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const isIsogram = (word) => {
  const acceptedStrings = word.toLowerCase().split('').filter(str => str >= 'a' && str <= 'z');
  return new Set(acceptedStrings).size === word.replace(/[-\s]/g, '').length;
};
